dashboard/client/api/kube-api.ts

```diff
@@ -32,20 +33,45 @@ export interface IKubeApiLinkRef {
   namespace?: string;
 }
 
-export class KubeApi<T extends KubeObject = any> {
-  static matcher = /(\/apis?.*?)\/(?:(.*?)\/)?(v.*?)(?:\/namespaces\/(.+?))?\/([^\/]+)(?:\/([^\/?]+))?.*$/
+export interface IKubeApiLinkBase extends IKubeApiLinkRef {
+  apiBase: string;
+  apiGroup: string;
+  apiVersionWithGroup: string;
+}
 
-  static parseApi(apiPath = "") {
+export class KubeApi<T extends KubeObject = any> {
+  static parseApi(apiPath = ""): IKubeApiLinkBase {
     apiPath = new URL(apiPath, location.origin).pathname;
-    const [, apiPrefix, apiGroup = "", apiVersion, namespace, resource, name] = apiPath.match(KubeApi.matcher) || [];
+    const [, prefix, ...parts] = apiPath.split("/");
+    const apiPrefix = `/${prefix}`;
+
+    const [left, right, found] = split(parts, "namespaces");
+    let apiGroup, apiVersion, namespace, resource, name;
+
+    if (found) {
+      if (left.length == 0) {
+        throw new Error(`invalid apiPath: ${apiPath}`)
```

Comment:
Throwing errors here is a breaking change (and it breaks the whole app in some cases).

Reply:
Well if this is throwing an error then that api would be a very good candidate for a new unit test and extending this function.